code/controllers/subsystem/runechat.dm

```diff
@@ -1,3 +1,14 @@
 TIMER_SUBSYSTEM_DEF(runechat)
 	name = "Runechat"
 	priority = FIRE_PRIORITY_RUNECHAT
+
+	var/list/datum/callback/message_queue = list()
+
+/datum/controller/subsystem/timer/runechat/fire(resumed)
+	. = ..() //poggers
+	while(message_queue.len)
+		var/datum/callback/queued_message = message_queue[message_queue.len]
+		queued_message.Invoke()
+		message_queue.len--
+		if(MC_TICK_CHECK)
+			return
```

Comment:
I feel like this should work like all other subsystems were they have 2 lists that are swapped around, allowing Cut() to be used instead of len-- and also making the queue FIFI rather than FILO